File: iosApp/iosApp/utils/AnalyticsBridge.m
#import <Foundation/Foundation.h>
#import <FirebaseAnalytics/FirebaseAnalytics.h>

// TODO simplify anyhow
@interface AnalyticsBridge : NSObject
+ (instancetype)shared;
- (void)logEvent:(NSString *)name parameters:(NSDictionary<NSString *, NSString *> *)parameters;
- (void)setUserId:(NSString *)userId;
- (void)setUserProperty:(NSString *)name value:(NSString *)value;
- (void)resetAnalyticsData;
@end

@implementation AnalyticsBridge

+ (instancetype)shared {
    static AnalyticsBridge *sharedInstance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[self alloc] init];
    });
    return sharedInstance;
}

- (void)logEvent:(NSString *)name parameters:(NSDictionary<NSString *, NSString *> *)parameters {
    [FIRAnalytics logEventWithName:name parameters:parameters];
}

- (void)setUserId:(NSString *)userId {
    [FIRAnalytics setUserID:userId];
}

- (void)setUserProperty:(NSString *)name value:(NSString *)value {
    [FIRAnalytics setUserPropertyString:value forName:name];
}

- (void)resetAnalyticsData {
    [FIRAnalytics resetAnalyticsData];
}

@end

File: iosApp/iosApp/utils/AnalyticsBridge.swift
import Foundation
import FirebaseAnalytics

@objc(AnalyticsBridge)
public class AnalyticsBridgeImpl: NSObject {
    @objc public static func shared() -> AnalyticsBridgeImpl {
        return AnalyticsBridgeImpl()
    }

    private let analytics: Analytics

    public override init() {
        self.analytics = Analytics.analytics()
        super.init()
    }

    @objc public func logEvent(_ name: String, parameters: [String: String]?) {
        analytics.logEvent(name, parameters: parameters)
    }

    @objc public func setUserId(_ userId: String?) {
        analytics.setUserID(userId)
    }

    @objc public func setUserProperty(_ name: String, value: String?) {
        analytics.setUserProperty(value, forName: name)
    }

    @objc public func resetAnalyticsData() {
        analytics.resetAnalyticsData()
    }
}
